classes/form/CustomerAddressForm.php

```diff
@@ -92,23 +92,29 @@ public function loadAddressById($id_address)
 
     public function fillWith(array $params = [])
     {
-        // This form is very tricky: fields may change depending on which
-        // country is being submitted!
-        // So we first update the format if a new id_country was set.
-        // If detect country from browser language is selected, default country will be overridden
-        if (Tools::isCountryFromBrowserAvailable()) {
-            $languageAvailable = Tools::getCountryIsoCodeFromHeader();
-            $this->formatter->setCountry(new Country(
-                (int) Country::getByIso($languageAvailable, true),
-                Language::getIdByIso($languageAvailable)
-            ));
-        } elseif (isset($params['id_country'])
-            && $params['id_country'] != $this->formatter->getCountry()->id
+        // This form is tricky: fields may change depending on which country is being selected!
+        // Country preselection priority order :
+        // 1) Update the format if a new id_country was set.
+        // 2) Detect country from browser language settings and matches BO enabled countries
+        // 3) Default country set in BO
+        if (isset($params['id_country']) && $params['id_country'] != $this->formatter->getCountry()->id) {
+            $this->formatter->setCountry(
+                new Country($params['id_country'], $this->language->id)
+            );
+        } elseif (
+            Tools::isCountryFromBrowserAvailable() &&
+            Country::getByIso($countryIsoCode = Tools::getCountryIsoCodeFromHeader(), true)
```

Comment:
I think it is confusing to add variable declaration inside a function call inside a elseif test ^^'

Reply:
I understand how it can be confusing. Still to me it is a valid syntax. And in this case we don't need to initiate a variable if the condition is not met. Two birds one stone. Saving memory. And comments help. But i understand Prestashop as an OSS, may want to make readability prevails against micro memory savings. Vote ? 😇

Reply:
I made a free comment, feel free to change or not ;)